First integration of Travis build stages

MMaiero opened the pull request:
Splitted kura build in two parts: Kura core first; distrib, example and tests in parallel
Added plugin to populate kura.build.version.
Removed stale wrl references in ant file.
Caching target-definition folder.
Build org.eclipse.kura-p2 before examples.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

Closes #1479

MMaiero committed:
First integration of Travis jobs

Splitted kura build in two parts: Kura core first; distrib, example and tests in parallel
Added plugin to populate kura.build.version.
Removed stale wrl references in ant file.
Caching target-definition folder.
Build org.eclipse.kura-p2 before examples.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

MMaiero commented:
@markoer

markoer commented on .travis.yml:
```diff
@@ -24,4 +29,17 @@ addons:
+    - stage: Main-build
+      script:
+        - $M2_HOME/bin/mvn -B -f target-platform/pom.xml clean install
+        - $M2_HOME/bin/mvn -B -f kura/pom.xml clean install 
```
See the comment at `pom.xml`.

markoer commented on kura/pom.xml:
```diff
@@ -69,7 +69,6 @@
         <module>org.eclipse.kura.wire.provider</module>
         <module>features</module>
         <module>emulator</module>
-        <module>test</module>
```
I don't like that this update involves removing tests by default.

If you do this, please add another profile, that is activated by default:
```
        <profile>
            <id>tests</id>
            <activation>
                <property>
                    <name>!skipTests</name>
                </property>
            </activation>
            <modules>
                <module>test</module>
            </modules>
        </profile>
```

Then, use the following script for the first stage:
`- $M2_HOME/bin/mvn -B -f kura/pom.xml clean install -DskipTests`

This way the tests would still start when building manually.

Similar should probably be done for the `org.eclipse.kura-p2` module as it now builds twice.

MMaiero replied:
My original idea was to add that step in the hudson build and in the build-all script. But we can do this way as well

MMaiero replied:
Same idea for the p2, even if its build will not be so much time consuming that we can decide to build twice

markoer replied:
That's true, P2 builds rather fast.

markoer reviewed (approved):
I really like what you did, here. But I suggest you take into account the comment related to running tests.

MMaiero commented:
I've added another commit to include Karaf in the parallel build and to clean the eclipse folder in .m2 in order to try to prevent possible errors with stored Kura artifacts. 
I've finally added the profile for tests.

MMaiero committed:
Added missing Karaf step to the build.

Added profile for tests in Kura pom.
Added action in first build to clean the m2 eclipse folder.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

MMaiero commented:
After this PR, I will provide another to format, a bit, the pom files, since they seem a little bit messy.
